packages/smooth_app/lib/pages/prices/price_proof_page.dart

```diff
@@ -119,11 +166,132 @@ class _PriceProofPageState extends State<PriceProofPage> {
       uriHelper: ProductQuery.uriPricesHelper,
     );
     if (prices.isError) {
+      setState(() {
+        _isLoading = false;
+      });
       return;
     }
-    _existingPrices = prices.value.items ?? <Price>[];
-    if (mounted) {
-      setState(() {});
+    setState(() {
+      _existingPrices = prices.value.items ?? <Price>[];
+      _isLoading = false;
+    });
+  }
+}
+
+/// Widget to display a grid of proofs with price count badges
+class ProofGridItem extends StatefulWidget {
```

Comment:
I may be old-fashioned but what is that code doing here in an unrelated file?

Assuming that this code is useful, it should be either in a specific file OR in a place where it makes sense, for instance where it's used, that is `prices_proofs_page.dart`, where there is a grid.